Error when trying to deploy a batch job

**Description**

Howdy,
I'm attempting to deploy a batch job using Levant, and Levant is crashing, I believe because it is trying to get the task group count from a batch job, which doesn't have such a thing.

**Relevant Nomad job specification file**

This is somewhat redacted but just names of things:

```
job "thing-periodic" {
  type = "batch"
  periodic {
    cron             = "*/5 * * * * *"
    prohibit_overlap = false
  }
  datacenters = ["us-west-2", "us-west-1", "us-west-1a", "us-west-1c", "us-west-2a", "us-west-2b", "us-west-2c"]
  group "thing-periodic" {
    task "thing-herd" {
      driver = "docker"
      config {
        image = "place.jfrog.io/thing:[[.DEPLOY_SUB___VERSION]]"
        command = "/usr/bin/python3"
        args = ["herd.py"]
        port_map = {}
        logging {
          type = "journald"
          config {
            tag = "${NOMAD_META_SPLUNK_INDEX}:${NOMAD_ALLOC_NAME}.${NOMAD_TASK_NAME}.${NOMAD_ALLOC_ID}"
          }
        }
      }
      resources {
        cpu = 200
        memory = 256
        network {
          mbits = 20
        }
      }
      env {
        CONFIG_FILE = "/local/herd.yml"
      }
      meta {
        SPLUNK_INDEX = "thing"
      }
      template {
        data = <<EOH
---
statsd_host: '{{ env "attr.driver.docker.bridge_ip" }}'
shepherds:{{ range service "thing@us-west-1" }}
- {{ .NodeAddress }}:{{ .Port }}{{ end }}{{ range service "thing@us-west-2" }}
- {{ .NodeAddress }}:{{ .Port }}{{ end }}
        EOH
        destination = "/local/herd.yml"
        change_mode = "noop"
      }
    }
    restart {
      attempts = 2
      delay = "15s"
      interval = "5m"
      mode = "fail"
    }
  }
}
```

**Output of `levant version`:**

```
Levant v0.0.3
```

**Output of `consul version`:**

```
Consul v0.7.5
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
```

**Output of `nomad version`:**

```
Nomad v0.7.0
```

**Additional environment details:**

Levant is running in our CD environment, which is GoCD on Ubuntu 14.04

**Debug log outputs from Levant:**

```
17:45:31.837 [INFO] levant/deploy: job thing-periodic not running, using template file group counts
17:45:31.839 panic: runtime error: invalid memory address or nil pointer dereference
17:45:31.839 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb2f968]
17:45:31.839 
17:45:31.840 goroutine 1 [running]:
17:45:31.840 github.com/jrasell/levant/levant.(*nomadClient).Deploy(0xc42000cb98, 0xc420083680, 0x0, 0xc42000cb00, 0x0)
17:45:31.840 	/home/travis/gopath/src/github.com/jrasell/levant/levant/deploy.go:70 +0xd8
17:45:31.840 github.com/jrasell/levant/command.(*DeployCommand).Run(0xc4201fb560, 0xc42000e0f0, 0x1, 0x6, 0xc4200c78a0)
17:45:31.840 	/home/travis/gopath/src/github.com/jrasell/levant/command/deploy.go:123 +0x5a7
17:45:31.840 github.com/jrasell/levant/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc420099a40, 0xc420099a40, 0x3, 0xc4200c7900)
17:45:31.840 	/home/travis/gopath/src/github.com/jrasell/levant/vendor/github.com/mitchellh/cli/cli.go:255 +0x1eb
17:45:31.840 main.RunCustom(0xc42000e090, 0x7, 0x7, 0xc4201fb410, 0xc4201fb350)
17:45:31.840 	/home/travis/gopath/src/github.com/jrasell/levant/main.go:49 +0x433
17:45:31.840 main.Run(0xc42000e090, 0x7, 0x7, 0xb37f08)
17:45:31.840 	/home/travis/gopath/src/github.com/jrasell/levant/main.go:17 +0x56
17:45:31.840 main.main()
17:45:31.841 	/home/travis/gopath/src/github.com/jrasell/levant/main.go:11 +0x63
```

I'm also getting the same issue:

```
[INFO] levant/deploy: job task-lead-summary-populate-user-delivery-times not running, using template file group counts
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb158ca]

goroutine 1 [running]:
github.com/jrasell/levant/levant.(*nomadClient).Deploy(0xc42000e640, 0xc4201eeb40, 0x0, 0xc42000e600, 0x0)
	/go/src/github.com/jrasell/levant/levant/deploy.go:70 +0xda
github.com/jrasell/levant/command.(*DeployCommand).Run(0xc4202adb00, 0xc42008e060, 0x1, 0x5, 0xc42000cfc0)
	/go/src/github.com/jrasell/levant/command/deploy.go:123 +0x5a7
github.com/jrasell/levant/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc4201f77c0, 0xc4201f77c0, 0x3, 0xc42000d020)
	/go/src/github.com/jrasell/levant/vendor/github.com/mitchellh/cli/cli.go:255 +0x1eb
main.RunCustom(0xc42008e010, 0x6, 0x6, 0xc4202ad9b0, 0xc4202ad8f0)
	/go/src/github.com/jrasell/levant/main.go:49 +0x433
main.Run(0xc42008e010, 0x6, 0x6, 0xb1e448)
	/go/src/github.com/jrasell/levant/main.go:17 +0x56
main.main()
	/go/src/github.com/jrasell/levant/main.go:11 +0x63
```

--- a/levant/deploy.go
+++ b/levant/deploy.go
@@ -61,20 +61,6 @@ func (c *nomadClient) Deploy(job *nomad.Job, autoPromote int, forceCount bool) (
 		}
 	}
 
-	// Check that the job has at least 1 TaskGroup with count > 0 (GH-16) if the
-	// job is not a System job. Systems jobs do not define counts so cannot be
-	// checked.
-	if *job.Type != nomadStructs.JobTypeSystem {
-		tgCount := 0
-		for _, group := range job.TaskGroups {
-			tgCount += *group.Count
-		}
-		if tgCount == 0 {
-			logging.Error("levant/deploy: all TaskGroups have a count of 0, nothing to do")
-			return
-		}
-	}
-
 	logging.Info("levant/deploy: triggering a deployment of job %s", *job.Name)
 
 	eval, _, err := c.nomad.Jobs().Register(job, nil)

